packages/admin/admin/src/form/file/FileSelectListItem.tsx

```diff
@@ -115,8 +117,12 @@ export const FileSelectListItem = (inProps: FileSelectListItemProps) => {
                             {ownerState.hasErrorWithoutDetails && (
                                 <ErrorIconContainer {...slotProps?.errorIconContainer}>{errorIcon}</ErrorIconContainer>
                             )}
-                            {Boolean(onClickDownload) && (
-                                <IconButton onClick={onClickDownload} disabled={disabled} {...slotProps?.iconButton}>
+                            {(Boolean(onClickDownload) || !!downloadUrl) && (
+                                <IconButton
+                                    disabled={disabled}
+                                    {...(downloadUrl ? { href: downloadUrl } : { onClick: onClickDownload })}
```

Comment:
Do we need to set the `download` on the `<a>` tag? Also, will this automatically be converted to an `<a>` tag?

Reply:
> Do we need to set the `download` on the `<a>` tag?

Good point, I added the `download` attribute. 

> Also, will this automatically be converted to an `<a>` tag?

Yes, MUIs `IconButton` automatically renders `a` instead of `button` when passing a `href`.